source/Octopus.Client/OctopusClientFactory.cs

```diff
@@ -33,7 +29,23 @@ public IOctopusClient CreateClient(OctopusServerEndpoint serverEndpoint)
         /// <returns>The <see cref="IOctopusAsyncClient" /> instance.</returns>
         public Task<IOctopusAsyncClient> CreateAsyncClient(OctopusServerEndpoint serverEndpoint, OctopusClientOptions octopusClientOptions = null)
         {
+            var requestingTool = DetermineRequestingTool();
             return OctopusAsyncClient.Create(serverEndpoint, octopusClientOptions, requestingTool);
         }
+
+        private string DetermineRequestingTool()
+        {
+            var launchAssembly = Assembly.GetEntryAssembly();
+            if (launchAssembly.GetTypes().Any(x => x.FullName == "Octo.Program"))
```

Comment:
This also needs to take into account our `dotnet-octo` global tool which has a `FullName` of `Octopus.DotNet.Cli.Program`.